main.cc

```diff
@@ -947,10 +948,14 @@ To start the scylla server proper, simply invoke as: scylla server (or just scyl
             }
             view_hints_dir_initializer.ensure_created_and_verified().get();
 
+            supervisor::notify("starting compaction_manager");
+            cm.start([&] { return make_compaction_manager_config(*cfg, dbcfg); }, std::ref(stop_signal.as_sharded_abort_source())).get();
+            auto stop_cm = deferred_stop(cm);
```

Comment:
the reason for lambda is to retrieve configuration from each shard, right? but reading make_compaction_manager_config() it looks like we don't have to. we're using the same parameters at all shards. could you please explain the motivation?

Reply:
@raphaelsc , I believe it's due to #7316

Reply:
We need to in order to get the per-shared updateable value in
https://github.com/scylladb/scylla/blob/39db15d2cba81870bcc8269ff8afebb2fe2d2038/utils/config_file.hh#L155-L158

Reply:
We have `sharded_parameter` for this purpose exactly.